src/mongodb/model.js

```diff
@@ -0,0 +1,290 @@
+import { isEmpty } from 'lodash/fp';
+import { formatMongodbErrors, createQueryRequiredError } from './errors';
+
+const streamEvents = {
```

Comment:
What does this do?

Reply:
It's a map of events emitted by `model.js`. 

*Why?*
Some Mongoose methods like `cursor` returns a stream. `model.js` handles the stream errors, formats the error and rethrows it. It can't re-emit an error event so emits a new event that clients of can access and use to log errors as needed